packages/zowe-explorer/src/utils/ProfilesUtils.ts

```diff
@@ -243,48 +285,171 @@ export class ProfilesUtils {
+                            false,
+                            vscode.ConfigurationTarget.Global
+                        );
+                    }
+                }
+            );
+        }
+    }
+
+    /**
+     * Prompts to install the missing VS Code extension associated with the credential manager override in imperative.json
+     *
+     * @param credentialManager the credential manager to handle its missing VS Code extension
+     * @returns Promise<void>
+     */
+    private static async promptAndHandleMissingCredentialManager(credentialManager: imperative.ICredentialManagerNameMap): Promise<void> {
+        ZoweLogger.trace("ProfilesUtils.promptAndHandleMissingCredentialManager called.");
+        const header = localize(
+            "ProfilesUtils.promptAndHandleMissingCredentialManager.suggestInstallHeader",
+            `Plugin of name "{0}" was defined for custom credential management on imperative.json file.`,
+            credentialManager.credMgrDisplayName
+        );
+        const installMessage = localize(
+            "ProfilesUtils.promptAndHandleMissingCredentialManager.suggestInstallMessage",
+            `Please install associated VS Code extension for
+        custom credential manager or revert to default.`
+        );
+        const revertToDefaultButton = localize("ProfilesUtils.promptAndHandleMissingCredentialManager.revertToDefault", "Use Default");
+        const installButton = localize("ProfilesUtils.promptAndHandleMissingCredentialManager.install", "Install");
+        await Gui.infoMessage(header, { items: [installButton, revertToDefaultButton], vsCodeOpts: { modal: true, detail: installMessage } }).then(
+            async (selection) => {
+                if (selection === installButton) {
+                    const credentialManagerInstallURL = vscode.Uri.parse(
+                        `https://marketplace.visualstudio.com/items?itemName=${credentialManager.credMgrZEName}`
+                    );
+                    if (await vscode.env.openExternal(credentialManagerInstallURL)) {
+                        const refreshMessage = localize(
+                            "ProfilesUtils.promptAndHandleMissingCredentialManager.refreshMessage",
+                            `After installing the extension, please make sure to reload
+                             your VS Code window in order to start using the installed credential manager`
+                        );
+                        const refreshButton = localize("ProfilesUtils.promptAndHandleMissingCredentialManager.refreshButton", "Refresh");
+                        if ((await Gui.showMessage(refreshMessage, { items: [refreshButton] })) === refreshButton) {
+                            await vscode.commands.executeCommand("workbench.action.reloadWindow");
+                        }
+                    }
+                }
+            }
+        );
+    }
+
+    public static async getProfileInfo(): Promise<imperative.ProfileInfo> {
```

Comment:
This is what I was referencing with ZE API.  We have a ProfilesCache.getProfileInfo() that makes the call to imperative to set the default credential manager.  If extenders like CICS are using this & their extension in an environment that would need the k8s ext for credentials shouldn't that be considered in the ProfilesCache.getProfileInfo() passing the correct information to imperative when an extenders calls the method.  If we don't they will always only get the default cred manager.

Reply:
Created issue https://github.com/zowe/vscode-extension-for-zowe/issues/2525 for follow up work on this, since we need to add an optional parameter that extenders can pass in for loading either the default or custom credential manager, aside from that this PR is good to go!